pkg/reconciler/tenancy/clusterworkspacedeletion/clusterworkspace_deletion_controller.go

```diff
@@ -50,7 +53,13 @@ const (
 	controllerName = "kcp-clusterworkspacedeletion"
 )
 
+var (
+	background        = metav1.DeletePropagationBackground
+	backgroudDeletion = metav1.DeleteOptions{PropagationPolicy: &background}
+)
+
 func NewController(
+	kubeClusterClient kubernetesclient.ClusterInterface,
```

Comment:
Shouldn't this be an `kubernetesclient.Interface` instead of the `kubernetesclient.ClusterInterface` , as it is done for the `kcpClusterClient` ?

Reply:
@davidfestal happy to update to `kubernetesclient.Interface` here -- just to make sure I understand why we would choose to do so: we are fine to use the `kubernetesclient.Interface` here rather than `kubernetesclient.ClusterInterface` because when we enqueue we are using a "cluster-aware" key (`key, err := kcpcache.MetaClusterNamespaceKeyFunc(obj)`), so specifying `.Cluster(lcn)` when performing operations in this controller would be redundant.

Does that sound right? Please feel free to correct anything I have incorrect. If folks would find it useful, I'd love to also add this convention to the developer docs.

Reply:
There are two cases: 

1. you are operating in a single-cluster context, as in a controller against a compute cluster, you use `cache.MetaNamespaceKeyFunc`, you have a `kubernetes.Interface`.
2. you are operating against kcp in a multi-cluster context, you use `kcpcache.MetaClusterNamespaceKeyFunc`, you have a `kubernetes.ClusterInterface`

Reply:
This is the second case, so let's keep the implementation the way it is.

Reply:
I might be missing some context here, but I don't see how it is a different use-case than the use of the `kcpClientCluster`. Could explain it ?